tutorials/haskell/generics/code/LICENSE.md

```diff
@@ -0,0 +1,21 @@
+MIT License
```

Comment:
Mark,
Thanks for writing this post!  It is really well written and structured.  I learned a lot editing it.  I have tried to understand Generics before and found myself with a headache trying to take in too much at once.  Saving the MetaData info for the end was a wise choice.
Most of my comments are small grammar fixes about missing articles, subjects an objects, e.g. a, the, we and us.  My biggest question is about this tricky `p` parameter.  It seems like that is a hard thing to explain, but I hope my questions in the comments below give you some inspiration for how to explain it just a little more clearly.
Cheers!
Nathan